File: frontend/src/types/skill.ts
export type ResourceRefType = "connector" | "mcp_server" | "knowledge_base" | "agent"

export interface ResourceRef {
  type: ResourceRefType
  id: string
  name: string
  alias: string
}

export interface SkillResponse {
  id: string
  user_id: string | null
  source?: string
  name: string
  description: string | null
  content: string
  script: string | null
  script_type: "python" | "shell" | null
  resource_refs: ResourceRef[] | null
  visibility: string
  org_id: string | null
  is_active: boolean
  status: string
  publish_status: string | null
  published_at: string | null
  reviewed_by: string | null
  reviewed_at: string | null
  review_note: string | null
  created_at: string
  updated_at: string | null
}

export interface SkillCreate {
  name: string
  description?: string | null
  content: string
  script?: string | null
  script_type?: "python" | "shell" | null
  resource_refs?: ResourceRef[] | null
  is_active?: boolean
}

export interface SkillUpdate {
  name?: string
  description?: string | null
  content?: string
  script?: string | null
  script_type?: "python" | "shell" | null
  resource_refs?: ResourceRef[] | null
  is_active?: boolean
}

export interface SkillTemplate {
  id: string
  name: string
  description: string
  icon?: string | null
  category: string
  blueprint: Record<string, unknown>
}

File: src/fim_one/web/schemas/skill.py
"""Skill request/response schemas."""

from __future__ import annotations

from typing import Any, Literal

from pydantic import BaseModel, Field


class SkillCreate(BaseModel):
    name: str = Field(min_length=1, max_length=200)
    description: str | None = None
    content: str = Field(default="")
    script: str | None = None
    script_type: Literal["python", "shell"] | None = None
    is_active: bool = True
    resource_refs: list[dict[str, Any]] | None = None


class SkillUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    content: str | None = None
    script: str | None = None
    script_type: Literal["python", "shell"] | None = None
    is_active: bool | None = None
    resource_refs: list[dict[str, Any]] | None = None


class SkillForkRequest(BaseModel):
    """Optional overrides when forking (cloning) a skill."""

    name: str | None = None  # Custom name; defaults to "{original} (Fork)"


class SkillResponse(BaseModel):
    id: str
    user_id: str | None
    name: str
    description: str | None
    content: str | None
    script: str | None
    script_type: str | None
    forked_from: str | None = None
    visibility: str = "personal"
    org_id: str | None = None
    is_active: bool = True
    status: str = "draft"
    publish_status: str | None = None
    published_at: str | None = None
    reviewed_by: str | None = None
    reviewed_at: str | None = None
    review_note: str | None = None
    source: str | None = None
    resource_refs: list[dict[str, Any]] | None = None
    created_at: str
    updated_at: str | None
